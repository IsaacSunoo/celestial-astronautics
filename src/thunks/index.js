import { setLoading, getApolloData, setError, getDaily } from '../actions';
import { fetchCall } from '../api';
import apiKey from '../api_key/apiKey';

// const baseUrl = 'https://api.nasa.gov/';

export const getApollo20 = () => {
    const url = 'https://images-api.nasa.gov/search?q=apollo%20landing&media_type=image';
    return async dispatch => {
        try {
            dispatch(setLoading(true));
            const apollo = await fetchCall(url);
            dispatch(setLoading(false));
            dispatch(getApolloData(apollo));
        } catch (err) {
            dispatch(setError(err.message));
        }
    }
}

export const getDailyNews = () => {
    const url = `https://api.nasa.gov/planetary/apod?api_key=${apiKey}`;
    return async dispatch => {
        try {
            dispatch(setLoading(true));
            const dailyNews = await fetchCall(url);
            dispatch(setLoading(false));
            dispatch(getDaily(dailyNews));
        } catch (err) {
            dispatch(setError(err.message));
        }
    }
}

