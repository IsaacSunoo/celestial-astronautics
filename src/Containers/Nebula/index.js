import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getNebula } from '../../thunks';
import LoadingGif from '../../images/Loading.gif';
import Particles from '../../components/Particles';
import InfoCard from '../../components/InfoCard';



class Nebula extends Component {
    componentDidMount = async () => {
        const { getNebula } = this.props;
        await getNebula();
    }

    render() {
        const { nebula, isLoading } = this.props;
        let displayNebula;
        if (nebula.length !== 0) {
            if (isLoading) {
                return (
                    <div className='loading-display'>
                        <h2>Loading</h2>
                        <img src={LoadingGif} alt='loading gif' />
                    </div>
                )
            } else {
                displayNebula = nebula.items.map(item => {
                    return (
                        <div key={item.data[0].nasa_id}>
                            <InfoCard information={item} />
                        </div>
                    )
                });
            }
        }

        return (
            <div className='info-items-container'>
                <Particles />
                {displayNebula}
            </div>
        )
    }
}

export const mapStateToProps = state => ({
    nebula: state.nebula,
    isLoading: state.isLoading
});

export const mapDispatchToProps = dispatch => ({
    getNebula: () => dispatch(getNebula())
});

export default connect(mapStateToProps, mapDispatchToProps)(Nebula);