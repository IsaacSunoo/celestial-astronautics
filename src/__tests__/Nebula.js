import React from 'react';
import { Nebula, mapStateToProps, mapDispatchToProps } from '../containers/Nebula';
import { getNebula } from '../thunks';
import { shallow } from 'enzyme';

jest.mock('../thunks');
const mockNebula = [];
const mockIsLoading = false;

describe('nebula', () => {
  let wrapper;
  window.fetch = jest.fn();
  beforeEach(() => {
    wrapper = shallow(<Nebula nebula={mockNebula} isLoading={mockIsLoading} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('returns an object with houses', () => {
      const mockState = {
        nebula: ['nebula', 'galactic'],
        isLoading: false,
        error: 'error'
      };
      const expectedState = {
        nebula: ['nebula', 'galactic'],
        isLoading: false
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expectedState);
    });
  });

  describe('mapDispatchToProps', () => {
    it('calls dispatch with a getNebula thunk', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = getNebula();
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.getNebula();
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});
