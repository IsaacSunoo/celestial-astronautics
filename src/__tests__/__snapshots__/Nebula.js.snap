// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nebula should match snapshot 1`] = `
<div
  className="info-items-container"
>
  <ParticlesContainer />
</div>
`;
